util/src/main/java/io/zeebe/util/health/CriticalComponentsHealthMonitor.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.util.health;
+
+import io.zeebe.util.sched.ActorControl;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+import org.slf4j.Logger;
+
+/** Healthy only if all components are healthy */
+public class CriticalComponentsHealthMonitor implements HealthMonitor {
+  private static final Duration HEALTH_MONITORING_PERIOD = Duration.ofSeconds(60);
+  private final Map<String, HealthMonitorable> monitoredComponents = new HashMap<>();
+  private final Map<String, HealthStatus> componentHealth = new HashMap<>();
+  private final AtomicReference<HealthStatus> healthStatus = new AtomicReference<>();
+  private final ActorControl actor;
+  private final Logger log;
+  private FailureListener failureListener;
+
+  public CriticalComponentsHealthMonitor(final ActorControl actor, final Logger log) {
+    this.actor = actor;
```

Comment:
Could it be an issue to run the monitor in the same an actor which is also doing something else (considering we have no supervisor actors yet)? I guess the `ZeebePartition` one doesn't really do much so it's probably safe there, but could it be an issue later?

Reply:
I think it will be an issue. Because this health monitor is monitoring sub components of this actor not the actor itself. And there is another monitor monitoring this actor. Do you see any problems?

Reply:
I was thinking, if we have a bad/slow actor, and it has to monitor/poll itself as well, then perhaps the "poll" never happens (e.g. the actor is stuck in a run-until-done loop)?